### ISSUE TITLE
Plugin doesn't notify of update from GitHub

### ISSUE THREAD

**Issue by LittleHamster** [YahnisElsts/plugin-update-checker#158]
Hi, 
I followed your GitHub(https://github.com/YahnisElsts/plugin-update-checker#github-integration) instructions and added this: 

```
require 'plugin-update-checker/plugin-update-checker.php';
$myUpdateChecker = Puc_v4p3_Factory::buildUpdateChecker(
	'https://github.com/my-user-name/my-repo-name/',
	__FILE__,
	'unique-plugin-slug'
);

//Optional: If you're using a private repository, specify the access token like this:
$myUpdateChecker->setAuthentication('my-token-here');

//Optional: Set the branch that contains the stable release.
$myUpdateChecker->setBranch('master');
```

to my plugin folder. because I have a private repo, I created an access token with these permissions: 
![image](https://user-images.githubusercontent.com/10330481/33813717-b0d12dd6-ddf3-11e7-9786-feb0f8a8d8fc.png)

I released version 1.0.0, then in master I released version 1.1.0 (written in plugin's header file) but the plugin doesn't detect the update. When I run "Check for update" it says "plugin is up to date.". 

Any ideas what is wrong here?

**Comment by ironicmoka** [comment]
If you downloaded the stable release the version the Readme file with instruction indicate the master version, you have to substitute the `Puc_v4p3_Factory` with `Puc_v4_Factory` in order to make it work. as from https://github.com/YahnisElsts/plugin-update-checker/issues/156

**Comment by YahnisElsts** [comment]
The class name probably isn't the issue in this case because if it was wrong then you would immediately get a fatal error ("class not found").

To get more information about the problem, install [Debug Bar](https://wordpress.org/plugins/debug-bar/) and look at the "PUC (you-slug)" panel. Does everything look correct? Click the "Request info" button. Does it display the correct version information or an error?

**Comment by LittleHamster** [comment]
@YahnisElsts 
Thanks for the quick reply, 
so here are the screenshots: 

![image](https://user-images.githubusercontent.com/10330481/33831780-74c63a58-de47-11e7-8957-c0db79293717.png)

![image](https://user-images.githubusercontent.com/10330481/33831853-af7add0c-de47-11e7-98ac-1c633d69914b.png)

so when I click Request Info, I get this: 
`Failed to retrieve plugin info from the metadata URL.
`

**Comment by YahnisElsts** [comment]
Please try the same thing with the PUC `master` branch. It has better error reporting and it should give you a more detailed error message.

**Comment by LittleHamster** [comment]
@YahnisElsts 
thanks for the quick reply, 
I updated to master and I got these errors when running "check now"
```
1) GitHub API error. Base URL: "/repos/:user/:repo/releases/latest", HTTP status code: 404.
2) GitHub API error. Base URL: "/repos/:user/:repo/tags", HTTP status code: 404.
3) GitHub API error. Base URL: "/repos/:user/:repo/branches/master", HTTP status code: 404.
4) Could not retrieve version information from the repository. This usually means that the update checker either can't connect to the repository or it's configured incorrectly.
```

so there is something wrong with the api token?
because this is a private repo, what kind of permissions should I give when creating the access token? because I want to distribute the plugin, I dont want to create a token that gives a lot of controls over other repos and stuff.

**Comment by LittleHamster** [comment]
here is what I used
![image](https://user-images.githubusercontent.com/10330481/33847079-6572aebc-de77-11e7-820b-e3c61658e30e.png)
 
when creating the personal access token, using these permissions plugin failed to update

I checked "Full control of private repositories" and I could update

the problem is that I want to distribute the plugin to other people and if they get the access token, they can access all my other private repos.

**Comment by YahnisElsts** [comment]
I've looked at the GitHub docs and I don't see an obvious way to give a token read-only access to the code in the repository. You can make read-only deploy keys, but that's not the same thing. It doesn't appear to be usable with the API.

I've seen suggestions that you could do something like this: transfer the repository to an organization, create a new user, add them to the organization and give them read-only access to the repository. Then create an access token as that user. It's pretty convoluted.

**Comment by abhijittricore** [comment]
I just want to know is there any way or setting available in git private repo 

where public users don't pull or push the repo using Token. I have used private repos for the update plugin checker 
where i need to pass my token to get the update notification from repos. i am afraid if someone 
finds my token then he is easily cloned or pushes the repos with code changes.

**Comment by YahnisElsts** [comment]
As was mentioned in an earlier comment, one way to prevent plugin users from pushing to the repository would be to create a new GitHub account and only give that account read-only access to the repository. If you create the token as that user, that token will also have read-only access.

As for someone cloning the repository, that might be an unavoidable risk when you use a repository to provide updates. If you want to avoid that, you might have to find a different way to distribute updates, like [wp-update-server](https://github.com/YahnisElsts/wp-update-server) or another self-hosted solution.
